src/nvim/charset.c

```diff
@@ -1739,8 +1739,11 @@ char_u* skiptowhite_esc(char_u *p) {
 /// @return Number read from the string.
 intmax_t getdigits(char_u **pp)
 {
+  errno = 0;
   intmax_t number = strtoimax((char *)*pp, (char **)pp, 10);
-  assert(errno != ERANGE);
```

Comment:
I haven't taken a look at the implementation of `strtoimax`, so I don't know if anything it calls could set errno.  The contract only guarantees that errno is valid to check if the return value is INTMAX_MAX or INTMAX_MIN.